### PR TITLE
CDAP-15822 Dynamic-schema-sinks plugin validation

### PR DESCRIPTION
**Pull request by mlozbin-cybervisiontech**
JIRA: https://issues.cask.co/browse/CDAP-15822
WIKI: https://wiki.cask.co/display/CE/Guidelines+on+Plugin+Validation

In scope of this PR:
- integrated with new validation API

### CODE REVIEW

**Comment by mlozbin-cybervisiontech** [comment]
@albertshau could you, please, take a look.

File: src/main/java/io/cdap/dynamicschema/DynamicSchemaHBaseSink.java [hunk 1]
```diff
@@ -251,6 +145,43 @@ public void transform(StructuredRecord input, Emitter<KeyValue<NullWritable, Mut
...
+      return;
+    }
+    // Check if the table exists on HBase.
+    try {
+      Configuration conf = HBaseConfiguration.create();
+      conf.set(HConstants.ZOOKEEPER_QUORUM, config.getQorum());
+      conf.setInt(HConstants.ZOOKEEPER_CLIENT_PORT, config.getClientPort());
+      HBaseAdmin admin = new HBaseAdmin(conf);
+      if (!admin.tableExists(config.getTable())) {
+        failureCollector.addFailure(String.format("HBase table '%s' does not exists", config.getTable()),
```
Comment (albertshau) [inline src/main/java/io/cdap/dynamicschema/DynamicSchemaHBaseSink.java]: exists -> exist

File: src/main/java/io/cdap/dynamicschema/DynamicSchemaHBaseSink.java [hunk 2]
```diff
@@ -251,6 +145,43 @@ public void transform(StructuredRecord input, Emitter<KeyValue<NullWritable, Mut
...
+                                    String.format("Create table '%s' using hbase shell", config.getTable()))
```
Comment (albertshau) [inline src/main/java/io/cdap/dynamicschema/DynamicSchemaHBaseSink.java]: this isn't always how people create tables, I think it's ok to have a null corrective action here.

File: src/main/java/io/cdap/dynamicschema/HBaseSinkConfig.java [hunk 3]
```diff
@@ -16,59 +16,72 @@
...
+  public static final String PORT = "port";
+  public static final String DURABILITY = "durability";
+  public static final String PATH = "path";
+
+  @Name(PORT)
+  @Description("Client port")
+  @Nullable
+  @Macro
+  private final String port;
+  @Name(DURABILITY)
```
Comment (albertshau) [inline src/main/java/io/cdap/dynamicschema/HBaseSinkConfig.java:56]: nit: separate variables with newlines

**Review by albertshau — COMMENTED** [review]
minor comments. Fix and squash, then i'll merge

File: src/main/java/io/cdap/dynamicschema/DynamicSchemaHBaseSink.java (hunk 1 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/io/cdap/dynamicschema/DynamicSchemaHBaseSink.java]: done

File: src/main/java/io/cdap/dynamicschema/DynamicSchemaHBaseSink.java (hunk 2 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/io/cdap/dynamicschema/DynamicSchemaHBaseSink.java]: done

File: src/main/java/io/cdap/dynamicschema/HBaseSinkConfig.java (hunk 3 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/io/cdap/dynamicschema/HBaseSinkConfig.java:56]: done

**Comment by mlozbin-cybervisiontech** [comment]
@albertshau I addressed all your comments and squashed commits, could you, please, merge.

**Review by albertshau — APPROVED** [review]

**Pull request merged**